Realm/RLMObject.h

```diff
@@ -250,6 +250,14 @@
  */
 + (NSArray *)ignoredProperties;
 
+/**
+ Implement to return an array of property names that should be required. These properties will not allow storing
```

Comment:
Implement to return an array of property names that should not allow storing `nil`.

By default, all properties of a type that support storing `nil` are considered optional properties. To require that an object in a Realm always have a non-`nil` value for a property, add the name of the property to the array returned from this method.

Currently only String, Data, and Object properties support storing `nil`, and all other properties are implicitly treated as if they were required properties. Support for for types will come in the future.